parmed/amber/parameters.py

```diff
@@ -94,7 +94,13 @@ def id_format(filename):
         is_fmt : bool
             True if it is an Amber-style parameter file. False otherwise.
         """
-        with closing(genopen(filename, 'r')) as f:
+        if isinstance(filename, string_types):
```

Comment:
You can't do this in `id_format`, since it consumes part of the file trying to determine if the file is a particular format, and not all file-like objects that get sent to id_format are necessarily seekable (websites, for example).